monkey/monkey_island/cc/dal/IRepository.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC
+from typing import Any, Optional, Sequence
+
+from monkey_island.cc.models import Config, Monkey
+from monkey_island.cc.models.edge import Edge
+
+
+class IRepository(ABC):
+
+    # Config
+    ###############################################
+
+    # This returns the current config
+    # TODO investigate if encryption should be here or where
+    def get_config(self) -> dict:
+        pass
+
+    def set_config(self, config: dict):
+        pass
+
+    # Used when only a subset of config is submitted, for example only PBAFiles
+    # Used by passing keys, like ['monkey', 'post_breach_actions', 'linux_filename']
+    # Using a list is less ambiguous IMO, than using . notation
+    def set_config_field(self, key_list: Sequence[str], value: Any):
+        pass
+
+    # Used when only a subset of config is needed, for example only PBAFiles
+    # Used by passing keys, like ['monkey', 'post_breach_actions', 'linux_filename']
+    # Using a list is less ambiguous IMO, than using . notation
+    # TODO Still in doubt about encryption, this should probably be determined automatically
+    def get_config_field(self, key_list: Sequence[str]) -> Any:
+        pass
+
+    # Edges
+    ###############################################
+
+    def get_all_edges(self):
+        pass
+
+    def get_edge(self, src_node_id: str, dst_node_id: str) -> Edge:
+        pass
+
+    def save_edge(self, edge: Edge):
+        pass
+
+    def get_by_dst_node(self, dst_node_id: str) -> Sequence[Edge]:
```

Comment:
We probably use both `dst` and `dest` in different places, we should pick one. Generally, "dnt abbrev", but I think "dest" is a common enough abbreviation that it's fine (same with "src"). I prefer "dest" to "dst". Thoughs?

Reply:
dst is more consistent with src, because both dont contain vowels. abbreviations usually don't contain vowels

Reply:
Ok. Document it in the guide.

Reply:
done